File: frontend/src/types/conversa.ts
export type StatusConversa = "ABERTA" | "PENDENTE" | "RESOLVIDA";
export type DirecaoMensagem = "ENTRADA" | "SAIDA";
export type StatusMensagem = "ENVIANDO" | "ENVIADA" | "ENTREGUE" | "LIDA" | "FALHOU" | "RECEBIDA";

export interface ContatoResumo {
  id: string;
  nome: string;
  telefone: string;
}

export interface Conversa {
  id: string;
  contato: ContatoResumo;
  responsavelUsuarioId: string | null;
  status: StatusConversa;
  ultimaMensagemEm: string;
  ultimaMensagemTexto: string | null;
  quantidadeNaoLidas: number;
}

export interface Mensagem {
  id: string;
  conversaId: string;
  direcao: DirecaoMensagem;
  tipo: string;
  texto: string | null;
  status: StatusMensagem;
  enviadaEm: string | null;
  recebidaEm: string | null;
  createdAt: string;
}

File: backend/WhatsCrm.Api/Entities/Conversa.cs
namespace WhatsCrm.Api.Entities;

public class Conversa
{
    public Guid Id { get; set; }
    public Guid EmpresaId { get; set; }
    public Guid ContatoId { get; set; }
    public Guid WhatsappContaId { get; set; }
    public Guid? ResponsavelUsuarioId { get; set; }
    public StatusConversa Status { get; set; } = StatusConversa.ABERTA;
    public DateTimeOffset UltimaMensagemEm { get; set; }
    public string? UltimaMensagemTexto { get; set; }
    public int QuantidadeNaoLidas { get; set; }
    public DateTimeOffset CreatedAt { get; set; }
    public DateTimeOffset UpdatedAt { get; set; }

    public Contato Contato { get; set; } = null!;
    public WhatsappConta WhatsappConta { get; set; } = null!;
    public Usuario? ResponsavelUsuario { get; set; }
    public ICollection<Mensagem> Mensagens { get; set; } = new List<Mensagem>();
}
